shardingsphere-proxy/shardingsphere-proxy-backend/src/main/java/org/apache/shardingsphere/proxy/backend/text/admin/mysql/executor/ShowTablesExecutor.java

```diff
@@ -78,7 +78,7 @@ private QueryResult getQueryResult(final String schemaName) {
     private Collection<String> getAllTableNames(final String schemaName) {
         Collection<String> allTableNames = ProxyContext.getInstance().getMetaData(schemaName).getSchema().getAllTableNames();
         if (showTablesStatement.getFilter().isPresent()) {
-            Optional<String> pattern = showTablesStatement.getFilter().get().getLike().map(each -> SQLUtil.convertLikePatternToRegex(each.getPattern()));
+            Optional<String> pattern = showTablesStatement.getFilter().get().getLike().map(each -> SQLUtil.convertLikePatternToRegex(each.getPattern().toLowerCase()));
             return pattern.isPresent() ? allTableNames.stream().filter(each -> each.matches(pattern.get())).collect(Collectors.toList()) : allTableNames;
```

Comment:
Hi @MingxingLAI, `allTableNames` returns lowercase table names now, which is the cause of confusion for users.

Reply:
I think there are three ways to slove this problem：

1. Treat all table names in lower case（like this pull request)
2. Distinguish case of table name
3. Ignore uppercase table names

Which solution do you think is the most proper ?

If we use the second solution, it may cause compatibility problems for many users who has use ShardingSphere and thread all table names in lower case. So, I this the first solution is proper.

Reply:
In addition，In the implement of ShardingSphere, treat all table names in lower case.

```java

public final class ShardingSphereSchema {
    
    private final Map<String, TableMetaData> tables;
    
    @SuppressWarnings("CollectionWithoutInitialCapacity")
    public ShardingSphereSchema() {
        tables = new ConcurrentHashMap<>();
    }
    
    public ShardingSphereSchema(final Map<String, TableMetaData> tables) {
        this.tables = new ConcurrentHashMap<>(tables.size(), 1);
        tables.forEach((key, value) -> this.tables.put(key.toLowerCase(), value));
    }
```

Reply:
@MingxingLAI I think we should distinguish case of table name for this issue. And the final result displayed to the user should be consistent with the table in the database, rather than converted to lowercase.

Reply:
I also think distinguish case of table name is more accurate, but I am still more worried about compatibility problem, or we invite more people to participate in this discussion?

Reply:
> In addition，In the implement of ShardingSphere, treat all table names in lower case.
> 
> ```java
> public final class ShardingSphereSchema {
>     
>     private final Map<String, TableMetaData> tables;
>     
>     @SuppressWarnings("CollectionWithoutInitialCapacity")
>     public ShardingSphereSchema() {
>         tables = new ConcurrentHashMap<>();
>     }
>     
>     public ShardingSphereSchema(final Map<String, TableMetaData> tables) {
>         this.tables = new ConcurrentHashMap<>(tables.size(), 1);
>         tables.forEach((key, value) -> this.tables.put(key.toLowerCase(), value));
>     }
> ```

@MingxingLAI Only the key is lowercase, and all the real information is stored in the values. So, I think we can use values to handle show tables logic.

Reply:
good idea, I will change the implement.

Reply:
Hi @MingxingLAI, does this pr has any progress?